packages/subgraph/src/plugin/admin.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  Action,
+  AdministratorAdminPlugin,
+  AdminProposal,
+  Administrator,
+} from '../../generated/schema';
+import {AdminMembers} from '../../generated/templates';
+import {
+  MembershipContractAnnounced,
+  ProposalCreated,
+  ProposalExecuted,
+} from '../../generated/templates/Admin/Admin';
+import {generateAdministratorAdminPluginEntityId} from '../utils/ids';
+import {
+  generateActionEntityId,
+  generatePluginEntityId,
+  generateProposalEntityId,
+} from '@aragon/osx-commons-subgraph';
+import {Address, dataSource, DataSourceContext} from '@graphprotocol/graph-ts';
+
+export function handleProposalCreated(event: ProposalCreated): void {
+  const context = dataSource.context();
+  const daoAddress = context.getString('daoAddress');
+  const metadata = event.params.metadata.toString();
+  _handleProposalCreated(event, daoAddress, metadata);
+}
+
+export function _handleProposalCreated(
+  event: ProposalCreated,
+  daoAddress: string,
+  metadata: string
+): void {
+  const pluginProposalId = event.params.proposalId;
+  const pluginAddress = event.address;
+  const pluginEntityId = generatePluginEntityId(pluginAddress);
+  const proposalEntityId = generateProposalEntityId(
+    pluginAddress,
+    pluginProposalId
+  );
+  const administratorAddress = event.params.creator;
+
+  const proposalEntity = new AdminProposal(proposalEntityId);
+  proposalEntity.daoAddress = Address.fromHexString(daoAddress);
+  proposalEntity.plugin = pluginEntityId;
+  proposalEntity.pluginProposalId = pluginProposalId;
+  proposalEntity.creator = administratorAddress;
+  proposalEntity.metadata = metadata;
+  proposalEntity.executed = false;
+  proposalEntity.createdAt = event.block.timestamp;
+  proposalEntity.startDate = event.params.startDate;
+  proposalEntity.endDate = event.params.endDate;
+  proposalEntity.administrator = administratorAddress.toHexString();
+  proposalEntity.allowFailureMap = event.params.allowFailureMap;
+  const administratorEntityId = generateAdministratorAdminPluginEntityId(
+    administratorAddress,
+    pluginAddress
+  );
+  let adminMemberEntity = AdministratorAdminPlugin.load(administratorEntityId);
+  if (!adminMemberEntity) {
+    adminMemberEntity = new AdministratorAdminPlugin(administratorEntityId);
+    adminMemberEntity.administrator = administratorAddress.toHexString();
+    adminMemberEntity.plugin = pluginEntityId;
+    adminMemberEntity.save();
+  }
+  let administratorEntity = Administrator.load(
+    administratorAddress.toHexString()
+  );
+  if (!administratorEntity) {
+    administratorEntity = new Administrator(administratorAddress.toHexString());
+    administratorEntity.address = administratorAddress.toHexString();
```

Comment:
I agree, I think we should remove the `address` and just let the `ID` to avoid having the same info twice. 

But, we wanted to have like a duplicate of the osx subgraph, as similar as possible. to avoid breaking the app (or others) in the transition, and removing a field could potentially be an issue. 

I still think we should avoid duplicating information, maybe we could create a following task for this. 

@Rekard0 wdyt?

Reply:
we have this anti pastern in couple of places,
these were demand by the UI, for filtering purposes
i would say for this iteration go the same way
but in this followup task we try to find a solution if any: https://aragonassociation.atlassian.net/browse/OS-1152